### PR TITLE
fix(am): addition of migration note for 4.0 about MongoDB indexes

### PR DESCRIPTION
**Pull request by leleueri**
related-to AM-63

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://graviteedocs.blob.core.windows.net/am-63-add-migrationguide/index.html)

### CODE REVIEW

File: pages/am/3.x/installation-guide/installation-guide-migration.adoc [hunk 1]
```diff
@@ -8,6 +8,15 @@ WARNING: If you plan to skip versions when you upgrade, ensure that you read the
 
 WARNING: Make sure to run scripts on the correct database since `gravitee-am` is not always the default database! Check your db name by running `show dbs`.
 
+== Upgrade to 4.0.0
+
+NOTE: For more information about the breaking changes of this version, see link:{{ '/am/current/am_breaking_changes_4.0.html' | relative_url }}[Breaking changes in 4.0]
+
+=== MongoDB indexes
+
+Starting from AM 3.21, the MongoDB indexes are now named using the first letter of the fields that compose the index. This change will allow to manage automatically indexes creation on DocumentDB. Before starting the Management API service, please execute the following link:https://github.com/gravitee-io/gravitee-access-management/blob/3.21.x/gravitee-am-repository/gravitee-am-repository-mongodb/src/main/resources/scripts/create-index.js[script] to delete indexes and recreate them with the right convention.
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/am/3.x/installation-guide/installation-guide-migration.adoc:17]: This link is broken

File: pages/am/3.x/breaking-changes/breaking-changes-4.0.adoc
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/am/3.x/breaking-changes/breaking-changes-4.0.adoc:1]: This file doesn't appear in the sidebar. Is this intentional?

File: pages/am/3.x/installation-guide/installation-guide-migration.adoc (hunk 1 above)
Comment (phiz71) [inline pages/am/3.x/installation-guide/installation-guide-migration.adoc:17]:
```suggestion
Starting from AM 4.0, the MongoDB indexes are now named using the first letter of the fields that compose the index. This change will allow to manage automatically indexes creation on DocumentDB. Before starting the Management API service, please execute the following link:https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-repository/gravitee-am-repository-mongodb/src/main/resources/scripts/create-index.js[script] to delete indexes and recreate them with the right convention.
```

Comment (phiz71) [reply pages/am/3.x/installation-guide/installation-guide-migration.adoc:17]: Could it be related to the fact that the repository is now private ?

File: pages/am/3.x/breaking-changes/breaking-changes-4.0.adoc
Comment (phiz71) [reply pages/am/3.x/breaking-changes/breaking-changes-4.0.adoc:1]: Question: for the 4.0 version, should we rename the `current` folder in `3.x` and start a new `current` folder in gravitee-docs ?
image

Comment (leleueri) [reply pages/am/3.x/breaking-changes/breaking-changes-4.0.adoc:1]: I asked myself the question but I considered that this folder "refactoring" is a task by itself and should be manage idependently. @AndrewMcFarlandCampbellGravitee are you agree ?

File: pages/am/3.x/installation-guide/installation-guide-migration.adoc (hunk 1 above)
Comment (AndrewMcFarlandCampbellGravitee) [reply pages/am/3.x/installation-guide/installation-guide-migration.adoc]: Probably.

Comment (leleueri) [reply pages/am/3.x/installation-guide/installation-guide-migration.adoc]: Link should be fixed by using the gh.gravitee.io proxy

**Review by loriepisicchio — APPROVED** [review]

**Pull request merged**